cirq-google/cirq_google/engine/virtual_engine_factory.py

```diff
@@ -367,3 +370,36 @@ def create_noiseless_virtual_engine_from_latest_templates() -> SimulatedLocalEng
     processor_ids = list(MOST_RECENT_TEMPLATES.keys())
     template_names = [MOST_RECENT_TEMPLATES[k] for k in processor_ids]
     return create_noiseless_virtual_engine_from_templates(processor_ids, template_names)
+
+
+def create_noisy_virtual_engine_from_processor_id_and_simulator_class(
+    processor_id: str, simulator_class: Type[SimulatesSamples], **kwargs
+) -> SimulatedLocalEngine:
+    """Creates a virtual engine with a noisy simulator based on a processor id.
+
+    Args:
+        processor_id: The string name of a processor that has available noise data.
+        simulator_class: The class of the type of simulator to be initialized.
+        kwargs: Other arguments which are passed through to the simulator initializer.
+            The 'noise' argument will be overwritten with a new noise model.
+
+    Returns:
+        A SimulatedLocalEngine that uses a simulator of type simulator_class with a
+            noise model based on available noise data for the processor processor_id.
+    """
+
+    calibration = load_median_device_calibration(processor_id)
+    noise_properties = noise_properties_from_calibration(calibration)
+    noise_model = NoiseModelFromGoogleNoiseProperties(noise_properties)
+    kwargs["noise"] = noise_model
+    simulator = simulator_class(**kwargs)
```

Comment:
I could be mistaken, but I think you can do `simulator_class(noise=noise_model, **kwargs)`

I think you also need a `#type: ignore` since the constructor is not specified by the interface.